src/test/scala/chiselTests/WarningSpec.scala

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.util._
+import chisel3.stage.{ChiselGeneratorAnnotation, ChiselStage}
+import chisel3.experimental.ChiselEnum
+import chiselTests.ChiselFlatSpec
+
+class WarningSpec extends ChiselFlatSpec with Utils {
+  behavior.of("Warnings")
+
+  "Warnings" should "be de-duplicated" in {
+    object MyEnum extends ChiselEnum {
+      val e0, e1, e2 = Value
+    }
+
+    class MyModule extends Module {
+      val in = IO(Input(UInt(2.W)))
+      val out1 = IO(Output(MyEnum()))
+      val out2 = IO(Output(MyEnum()))
+      out1 := MyEnum(in); out2 := MyEnum(in)
```

Comment:
```suggestion
      def func(out: MyEnum): Unit = {
        out := MyEnum(in)
      }
      func(out1)
      func(out2)
```
While the two warnings on the same line are deduplicated (due to only line number being provided in the source locator here), that may not always be the case (if we were to start using `SourceInfo` in warnings). More importantly though, I think it doesn't show the intended use case which is warnings that do point to the same exact thing which more often comes up from calling a utility multiple times.